### DISCUSSION TITLE
Ender 3 S1 F4 Filament Runout Sensor

### DISCUSSION THREAD

**Discussion by cody-brockman** [mriscoc/Ender3V2S1#1198] [Q&A] [answered]
I'm still new to figuring out how the professional firmware works... 

I am trying to get the runout sensor to work with this firmware (Ender3S1-F4-20231212). The sensor was working on the stock Creality firmware. 

I have a stock Ender 3 S1 F4 with the filament sensor that came with the printer. I have found the "filament settings" and have tried enabling the runout. I have tried the High, Low, and Motion options. I have also changed the runout distance to 0. I'm sure that one of these will work, I'm just not sure which combination I need for my printer. 

I can go to the endstop diagnostics and see triggered and open when inserting filament into it. But I'm not sure if triggered or open means there is filament in the runout sensor and which means there is not filament in the sensor.

**Comment by cody-brockman** [comment, marked as answer]
Just in case someone else is in my same position…

I think I have it figured out. 

When testing the filament sensor. “Triggered” means there is filament in the sensor. “Open” means there is no filament in it. 

I enabled the sensor. Then turned the setting to “high”. It appears to be working now. We’ll see in the morning.

**Comment by mriscoc** [comment]
Check the documentation: https://github.com/mriscoc/Ender3V2S1/wiki/Filament-Settings-Menu#run-out-sensor
